libs/dyn/convert/from_typed.go

```diff
@@ -41,6 +41,14 @@ func fromTyped(src any, ref dyn.Value, options ...fromTypedOptions) (dyn.Value,
 			return dyn.NilValue, nil
 		}
 		srcv = srcv.Elem()
+
+		// If a pointer to a scalar type points to a zero value, we should include
+		// that zero value in the dynamic representation.
+		// This is because by default a pointer is nil in Go, and it not being nil
+		// indicates its value was intentionally set to zero.
+		if !slices.Contains(options, includeZeroValues) {
+			options = append(options, includeZeroValues)
+		}
```

Comment:
This has a side-effect of enabling this option for all nested structs.

With resources being a map of string to a pointer type, I expect this to include all zero-values for all resources.

Reply:
Today the option is only passed fromTyped(Scalar) methods. These methods are also the ones that implement `includeZeroValues` so this would be not an issue. The options will get lost for structs during recursion.

I added a isScalar check though to make this more robust.

Reply:
TODO: Also clarifying the language around the flag a bit, it's a bit confusing right now it seems.

Reply:
Done. The comments and semantics associated with the flag are clearer now. I also added tests to assert that the nested struct case you mentioned is a non-issue.